Apps/W1/File - SharePoint Connector/app/src/SharePointConnectorImpl.Codeunit.al

```diff
@@ -0,0 +1,444 @@
+        MarkerFileNameTok: Label 'BusinessCentral.FileSystem.txt', Locked = true;
+        NotFoundTok: Label '404', Locked = true;
+
+    /// <summary>
+    /// Gets a List of Files stored on the provided account.
+    /// </summary>
+    /// <param name="AccountId">The file account ID which is used to get the file.</param>
+    /// <param name="Path">The file path to list.</param>
+    /// <param name="FilePaginationData">Defines the pagination data.</param>
+    /// <param name="Files">A list with all files stored in the path.</param>
+    procedure ListFiles(AccountId: Guid; Path: Text; FilePaginationData: Codeunit "File Pagination Data"; var FileAccountContent: Record "File Account Content" temporary)
+    var
+        SharePointFile: Record "SharePoint File";
+        SharePointClient: Codeunit "SharePoint Client";
+        OrginalPath: Text;
+    begin
+        OrginalPath := Path;
+        InitPath(AccountId, Path);
+        InitSharePointClient(AccountId, SharePointClient);
+        if not SharePointClient.GetFolderFilesByServerRelativeUrl(Path, SharePointFile) then
+            ShowError(SharePointClient);
+
+        FilePaginationData.SetEndOfListing(true);
+
+        if not SharePointFile.FindSet() then
+            exit;
+
+        repeat
+            FileAccountContent.Init();
+            FileAccountContent.Name := SharePointFile.Name;
+            FileAccountContent.Type := FileAccountContent.Type::"File";
+            FileAccountContent."Parent Directory" := OrginalPath;
+            FileAccountContent.Insert();
+        until SharePointFile.Next() = 0;
+    end;
+
+    /// <summary>
+    /// Gets a file from the provided account.
+    /// </summary>
+    /// <param name="AccountId">The file account ID which is used to get the file.</param>
+    /// <param name="Path">The file path inside the file account.</param>
+    /// <param name="Stream">The Stream were the file is read to.</param>
+    procedure GetFile(AccountId: Guid; Path: Text; Stream: InStream)
+    var
+        SharePointFile: Record "SharePoint File";
+        SharePointClient: Codeunit "SharePoint Client";
+        TempBlob, TempBlob2 : Codeunit "Temp Blob";
```

Comment:
TempBlob2 is unused.

@IceOnly  regarding your issue with the empty stream.
Take a look at the following issue: [Stream](https://github.com/microsoft/AL/issues/7329) 

Why aren't you directly using the parameter `Stream` in the procedure `DownloadFileContentByServerRelativeUrl`?